### PR TITLE
Merging sell-stock to main 

### PR DESCRIPTION
**Pull request by Peaceful36**
I've implemented my classess based on the clean architecture but still need to add some test files to my user story.

### CODE REVIEW

File: src/main/java/SellStockUseCase/SellUseCaseInteractor.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+     * @param sell The request object containing the portfolio, symbol, and quantity
+     * @return The response object containing the portfolio, symbol, quantity, and price
+     * @throws IOException
+     */
+
+    public SellOutputResponse sellStock(SellInputRequest sell) throws IOException {
+        Portfolio portfolio = sell.getPortfolio();
+        String symbol = sell.getSymbol();
+        int quantity = sell.getQuantity();
+        boolean possible = portfolio.sellStock(symbol, quantity);
```
Comment (Spaghetti-12) [inline src/main/java/SellStockUseCase/SellUseCaseInteractor.java:21]: portfolio.sellStock takes a 3rd parameter, a String representing the username.

**Comment by Rohh1** [comment]
Looks good. Passing in the username parameter will definitely help with keeping track of which account is selling what.

File: src/main/java/SellStockUseCase/SellUseCaseInteractor.java (hunk 1 above)
Comment (Peaceful36) [reply src/main/java/SellStockUseCase/SellUseCaseInteractor.java]: Fixed!

File: src/main/java/SellStockUseCase/SellStockController.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+import java.io.IOException;
+public class SellStockController {
+    private final SellUseCaseInteractor interactor;
+
+    public SellStockController() {
+        interactor = new SellUseCaseInteractor();
+    }
+
+    public SellOutputResponse sellStock(SellInputRequest sell) throws IOException {
+        SellUseCaseInteractor sellUseCaseInteractor = new SellUseCaseInteractor();
```
Comment (eladperlman) [inline src/main/java/SellStockUseCase/SellStockController.java:12]: what's the point of making another interactor, u already instantiate one in the constructer

Comment (Peaceful36) [reply src/main/java/SellStockUseCase/SellStockController.java]: Don't know why I did that while coding. I just changed it.

**Review by Rohh1 — DISMISSED** [review]
Looks good!

**Review by Spaghetti-12 — DISMISSED** [review]

**Review by inaam10-tech — APPROVED** [review]

**Review by Spaghetti-12 — APPROVED** [review]

**Review by noel-toms — APPROVED** [review]
The code looks fine, but I would recommend adding comments and descriptions in each function. They don't have to be detailed but should be informative enough for someone to understand the functionality.

**Comment by Peaceful36** [comment]
I will add some tests and code comments to make my thing clearer.

**Comment by inaam10-tech** [comment]
Have you added testing yet?

File: src/main/java/SellStockUseCase/SellStockController.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package SellStockUseCase;
+
+import java.io.IOException;
+public class SellStockController {
+
+
+    public SellStockController() {
```
Comment (noel-toms) [inline src/main/java/SellStockUseCase/SellStockController.java:7]: What is the point of the empty constructor?

**Pull request merged**